internal/tiltfile/tiltfile_state.go

```diff
@@ -616,6 +616,34 @@ func (s *tiltfileState) assembleK8sV2() error {
 				r.name = opts.newName
 				s.k8sByName[r.name] = r
 			}
+
+			selectors := make([]k8sObjectSelector, len(opts.objects))
+			for i, o := range opts.objects {
+				s, err := selectorFromString(o)
+				if err != nil {
+					return errors.Wrap(err, "Error making selector from string")
+				}
+				selectors[i] = s
+			}
+
+			for _, selector := range selectors {
+				matches := []k8s.K8sEntity{}
+				for _, e := range s.k8sUnresourced {
+					if selector.matches(e) {
+						matches = append(matches, e)
+					}
+				}
+				numMatches := len(matches)
+				if numMatches == 0 {
+					return fmt.Errorf("Unable to find any unesourced matches for selector %+v", selector)
```

Comment:
If this is going to be a user-facing error message, i think in the UI etc. we don't use the language "unresourced" -- either say "uncategorized" or (ideally) be more specific: "selectors passed to k8s_resource.objects must match exactly one k8s entity not already part of a Tilt resource" or something